onnxruntime/core/providers/migraphx/migraphx_allocator.h

```diff
@@ -11,27 +11,27 @@ namespace onnxruntime {
 
 class MIGraphXAllocator : public IAllocator {
  public:
-  MIGraphXAllocator(int device_id, const char* name)
+  MIGraphXAllocator(const OrtDevice::DeviceId device_id, const char* name)
```

Comment:
You do not need to use const here.  See https://abseil.io/tips/109 and https://google.github.io/styleguide/cppguide.html#Use_of_const

Reply:
Then I think we do not have good clang-tidy config here.